presto-main/src/main/java/com/facebook/presto/sql/planner/ExpressionInterpreter.java

```diff
@@ -1239,7 +1241,23 @@ public static Object invoke(ConnectorSession session, ScalarFunctionImplementati
             handle = handle.bindTo(session);
         }
         try {
-            return handle.invokeWithArguments(argumentValues);
+            List<Object> actualArguments = new ArrayList<>();
+            Class<?>[] parameterArray = function.getMethodHandle().type().parameterArray();
+            for (int i = 0; i < argumentValues.size(); i++) {
+                Object argument = argumentValues.get(i);
+                if (function.getNullFlags().get(i)) {
+                    boolean isNull = argument == null;
+                    if (isNull && !Primitives.isWrapperType(parameterArray[actualArguments.size()])) {
```

Comment:
Same here.

BTW, we should probably migrate this code to use FunctionInvoker, but that's a separate change.